climada/hazard/base.py

```diff
@@ -157,49 +158,88 @@ class Hazard():
     """Name of the variables that aren't need to compute the impact. Types:
     scalar, string, list, 1dim np.array of size num_events."""
 
-    def __init__(self, haz_type='', pool=None):
-        """
-        Initialize values.
+    def __init__(self,
+                 haz_type: str,
+                 intensity: sparse.csr_matrix,
+                 centroids: Centroids,
+                 date: np.ndarray,
+                 units: str = "",
+                 frequency: Optional[np.ndarray] = None,
+                 frequency_unit: str = DEF_FREQ_UNIT,
+                 orig: Optional[np.ndarray] = None,
+                 event_id: Optional[np.ndarray] = None,
+                 event_name: Optional[List[str]] = None,
+                 fraction: Optional[sparse.csr_matrix] = None,
+                 pool: Optional[Pool] = None,
+                 fast: bool = False,
```

Comment:
I am not convinced by the addition of this variable. We have this no-where in the code, and I also have not seen that in other Python modules.

Reply:
Is your issue the existence of the variable or its name? With the full initialization of the object we now have the chance to perform an automated check of the resulting object. I think this is very important and should only be skipped if a user is absolutely sure that the data is consistent.

I don't think the current checks are very expensive. So if you opt for removing this variable, I would be in favor of *always* performing the check.

Reply:
I agree. This is the route we took so far with `Impact`.